### PR TITLE
SDK support for Content Versioning

### LINKED ISSUE #19999: Add SDK support for Content Versioning

### PR DESCRIPTION
**Pull request by azrikahar**
## Scope

What's changed:

- added `directus_versions` schema
- added CRUD operations for Content Versions
- added utils for Content Versions (compare, save, and promote)
- Update every SDK examples in Content Versions API reference

## Potential Risks / Drawbacks

- Perhaps a tiny increase of the SDK's codebase

## Review Notes / Questions

- May require more attention in `rest/commands/utils/versions.ts` in terms of the typings used, as I'm not sure did I form them correctly

---

Fixes #19999

### CODE REVIEW

File: sdk/src/rest/commands/utils/versions.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+/**
+ * Compare an existing version with the main item.
+ *
+ * @param id Primary key of the version.
+ *
+ * @returns All fields with different values, along with the hash of the main item and the information whether the current
+version is outdated (main item has been updated since the creation of the current version)
+ */
+export const compareVersion =
```
Comment (br41nslug) [inline sdk/src/rest/commands/utils/versions.ts:13]: I feel the naming here is too vague and possibly even misleading. lets use a more feature specific and descriptive name here for these SDK functions like `compareDraftVersion` or `compareRevisionVersion`.

Comment (azrikahar) [reply sdk/src/rest/commands/utils/versions.ts:13]: Makes sense, `compareContentVersion` sounds fitting with Content Versioning 👍 In that case, I'll proceed with updating all of them such as `readVersion`/`updateVersion` to `readContentVersion`/`updateContentVersion` respectively as well.

Comment (azrikahar) [reply sdk/src/rest/commands/utils/versions.ts]: resolved via https://github.com/directus/directus/pull/20000/commits/e71199bf9555eb164c374ddbecd433d81ad6de15 🙌

File: sdk/src/rest/commands/utils/versions.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+			mainHash: string;
+			current: Partial<Item>;
+			main: Item;
+		},
+		Schema
+	> =>
+	() => ({
+		path: `/versions/${id}/compare`,
+		method: 'GET',
+	});
```
Comment (br41nslug) [inline sdk/src/rest/commands/utils/versions.ts:28]: I think we may want to check for empty `id` here too so we don't get `/versions//compare`
```suggestion
	> =>
	() => {
		throwIfEmpty(key, 'Key cannot be empty');
		
	    return {
    		path: `/versions/${id}/compare`,
	    	method: 'GET',
	    }
	};
```

File: sdk/src/rest/commands/utils/versions.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+export const saveContentVersion =
+	<Schema extends object, Collection extends keyof Schema, Item = UnpackList<Schema[Collection]>>(
+		id: DirectusVersion<Schema>['id'],
+		item: Partial<Item>
+	): RestCommand<Item, Schema> =>
+	() => ({
+		path: `/versions/${id}/save`,
+		method: 'POST',
+		body: JSON.stringify(item),
+	});
```
Comment (br41nslug) [inline sdk/src/rest/commands/utils/versions.ts:47]: Same ID check for empty here would be good

File: sdk/src/rest/commands/utils/versions.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+	<Schema extends object, Collection extends keyof Schema, Item = UnpackList<Schema[Collection]>>(
+		id: DirectusVersion<Schema>['id'],
+		mainHash: string,
+		fields?: (keyof UnpackList<Item>)[]
+	): RestCommand<string | number, Schema> =>
+	() => ({
+		path: `/versions/${id}/promote`,
+		method: 'POST',
+		body: JSON.stringify(fields ? { mainHash, fields } : { mainHash }),
+	});
```
Comment (br41nslug) [inline sdk/src/rest/commands/utils/versions.ts:68]: Same ID check for empty here would be good

**Review by br41nslug — DISMISSED** [review]
Couple of small tweaks but the types and docs are looking great 🚀

File: sdk/src/rest/commands/utils/versions.ts (hunk 2 above)
Comment (paescuj) [reply sdk/src/rest/commands/utils/versions.ts]: Great catch ❤️

**Review by paescuj — APPROVED** [review]

**Pull request merged**